src/langchain_google_spanner/vector_store.py

```diff
@@ -832,15 +967,111 @@ def similarity_search_with_score_by_vector(
             List[Document]: List of documents most similar to the query.
         """
 
-        results, column_order_map = self._get_rows_by_similarity_search(
+        results, column_order_map = self._get_rows_by_similarity_search_knn(
             embedding, k, pre_filter
         )
         documents = self._get_documents_from_query_results(
             list(results), column_order_map
         )
         return documents
 
-    def _get_rows_by_similarity_search(
+    def search_by_ANN(
+        self,
+        column_name: str,
+        table_name: str,
+        index_name: str,
```

Comment:
You don't need to take it from the user... as it's defined in SpannerVectorStore at the time of initialization.